sources/packages/api/src/services/application/application.service.ts

```diff
@@ -675,35 +685,133 @@ export class ApplicationService extends RecordDataModelService<Application> {
-      })
-      .orderBy("application.applicationNumber", "DESC")
-      .getMany();
+      });
+
+    activeApplicationQuery.andWhere("application.isArchived = :isArchived", {
+      isArchived: archived,
+    });
+
+    if (archived) {
+      activeApplicationQuery.orWhere(
+        new Brackets((qb) => {
+          qb.where("studentScholasticStanding.id IS NOT NULL");
+        }),
+      );
+    }
+
+    activeApplicationQuery.orderBy("application.applicationNumber", "DESC");
+
+    if (paginationOptions.searchCriteria) {
+      activeApplicationQuery
+        .andWhere(
+          new Brackets((qb) => {
+            qb.where(
+              "CONCAT(user.firstName,' ', user.lastName) Ilike :searchCriteria",
+            ).orWhere("application.applicationNumber Ilike :searchCriteria");
+          }),
+        )
+        .setParameter(
+          "searchCriteria",
+          `%${paginationOptions.searchCriteria.trim()}%`,
+        );
+    }
+    activeApplicationQuery
+      .orderBy(
+        this.transformToEntitySortField(
+          paginationOptions.sortField,
+          paginationOptions.sortOrder,
+        ),
+      )
+      .offset(paginationOptions.page * paginationOptions.pageLimit)
+      .limit(paginationOptions.pageLimit);
+
+    const [result, count] = await activeApplicationQuery.getManyAndCount();
+    return {
+      results: result,
+      count: count,
+    };
+  }
+
+  /**
+   * Returns application status with respect to archive status and scholastic standing change.
+   * @param isArchived archive status.
+   * @param studentScholasticStandingId scholastic standing id.
+   * @returns application scholastic standing status.
+   */
+  getApplicationSholasticStandingStatus(
+    isArchived: boolean,
+    studentScholasticStandingId?: number,
+  ): ApplicationSholasticStandingStatus {
+    if (isArchived) {
+      return ApplicationSholasticStandingStatus.Unavailable;
+    }
+    if (studentScholasticStandingId) {
+      return ApplicationSholasticStandingStatus.Completed;
+    }
+    return ApplicationSholasticStandingStatus.Available;
+  }
+
+  /**
+   * Transformation to convert the data table column name to database column name.
+   * Any changes to the data object (e.g data table) in presentation layer must be adjusted here.
+   * @param sortField database fields to be sorted.
+   * @param sortOrder sort order of fields (Ascending or Descending order).
+   * @returns OrderByCondition
+   */
+  private transformToEntitySortField(
+    sortField: string,
+    sortOrder: FieldSortOrder,
+  ): OrderByCondition {
+    const orderByCondition = {};
+    if (sortField === "fullName") {
+      orderByCondition["user.firstName"] = sortOrder;
+      orderByCondition["user.lastName"] = sortOrder;
+      return orderByCondition;
+    }
+
+    const fieldSortOptions = {
+      applicationNumber: "application.applicationNumber",
+    };
+    const dbColumnName = fieldSortOptions[sortField];
```

Comment:
This logic is not handling the default sort if the sort field is not provided.

If the sort field is null(it can be null as it is Query Param) then the orderByCondition will be `{null:"ASC|DESC"}`

also the there is no default sort order. In this case orderByCondition will be `{null:null}`

I would recommend setting default sort values like this. 
![image](https://user-images.githubusercontent.com/54600590/174923510-b5f5844b-161c-48f7-9b78-97c112bbc38d.png)